### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by Holger-Will**

### CODE REVIEW

**Comment by Holger-Will** [comment]
#29

**Comment by Holger-Will** [comment]
maybe it's better to use new Error(msg)

File: test/ESP3Parser.test.js [hunk 1]
```diff
@@ -45,7 +45,9 @@ describe('serialport enocean parser', function () {
        */
       it('packets MUST be emitted, if messy bytes occur before the header was detected and there are at least 5 bytes to real sync byte', function () {
         const spy = sinon.spy()
+        const error_spy = sinon.spy()
```
Comment (RafaelKa) [inline test/ESP3Parser.test.js]: Move to separate test case
```suggestion
```

File: test/ESP3Parser.test.js [hunk 2]
```diff
@@ -45,7 +45,9 @@ describe('serialport enocean parser', function () {
...
         esp3parser.on('data', spy)
+        esp3parser.on('error', error_spy)
```
Comment (RafaelKa) [inline test/ESP3Parser.test.js]: Move to separate test case
```suggestion
```

File: test/ESP3Parser.test.js [hunk 3]
```diff
@@ -65,6 +67,7 @@ describe('serialport enocean parser', function () {
 
         esp3parser.write(largeAndMessyByteStream)
         assert.equal(spy.callCount, telegrams.length, 'Received unexpected count of packets.')
+        assert.equal(error_spy.callCount, telegrams.length, 'Received unexpected count of packets.')
```
Comment (RafaelKa) [inline test/ESP3Parser.test.js]: Move to separate test case
```suggestion
```

File: test/ESP3Parser.test.js [hunk 4]
```diff
@@ -65,6 +67,7 @@ describe('serialport enocean parser', function () {
...
       })
```
Comment (RafaelKa) [inline test/ESP3Parser.test.js]: Move to separate test case
```suggestion

      it('error must be emitted on failed CRC8 checks for ESP3 Packet header', function () {
        const spy = sinon.spy()
        esp3parser.on('error', spy)
        esp3parser.write(Buffer.from('55af600ffd9b7812001f', 'hex'))
        assert.isTrue(spy.calledOnce, 'Error is not emitted on fails in ESP3 Packet header.')
        assert.equal(spy.getCall(0).args[0].code, 1, 'Wrong error code on failing CRC8 check for header.')
      })

      it('error must be emitted on failed CRC8 Checks for ESP3 Packet datas', function () {
        const spy = sinon.spy()
        esp3parser.on('error', spy)
        esp3parser.write(Buffer.from('55000a0701eba540300287ffd9b7e50001ffffffff440000', 'hex'))
        assert.isTrue(spy.calledOnce, 'Error is not emitted on fails in ESP3 Packet header.')
        assert.equal(spy.getCall(0).args[0].code, 2, 'Wrong error code on failing CRC8 check for datas.')
      })
```

**Comment by RafaelKa** [comment]
> maybe it's better to use new Error(msg)

new Error(string) sucks! Object incl. error code is Top :+1: 

separate test case and different error codes for both, then can this land onto master

File: ESP3Parser.js [hunk 5]
```diff
@@ -96,8 +100,12 @@ class ESP3Parser extends Transform {
       [this.currentESP3Packet.data, this.currentESP3Packet.optionalData],
       (this.currentESP3Packet.header.dataLength + this.currentESP3Packet.header.optionalLength)
     )
-    if (this.getCrc8(datas) !== byte) {
-      // todo : Log this fail
+    if (this.getCrc8(datas) != byte) {
+      this.emit('error', {
+        code: 1,
```
Comment (RafaelKa) [inline ESP3Parser.js]:
```suggestion
        code: 2,
```

**Pull request merged**